src/main/protobuf/soc/moves.proto

```diff
@@ -0,0 +1,141 @@
+syntax = "proto2";
+
+package protos.soc;
+
+import "soc/inventory.proto";
+import "soc/board.proto";
+import "soc/state.proto";
+
+enum ActionType {
+    ROLL_DICE = 0;
+    END_TURN = 1;
+    INITIAL_PLACEMENT = 2;
+    DISCARD_RESOURCES = 3;
+    MOVE_ROBBER_AND_STEAL = 4;
+    BUY_DEVELOPMENT_CARD = 5;
+    BUILD_ROAD = 6;
+    BUILD_SETTLEMENT = 7;
+    BUILD_CITY = 8;
+    PORT_TRADE = 9;
+    PLAY_KNIGHT = 10;
+    PLAY_YEAR_OF_PLENTY = 11;
+    PLAY_MONOPOLY = 12;
+    PLAY_ROAD_BUILDER = 13;
+}
+
+message HiddenCard {
+    repeated Player viewable_by = 1;
+    oneof card {
+        Resource resource_card = 2;
+        DevelopmentCard development_card = 3;
+    }
+}
+
+message ActionSpecification {
+    required ActionType type = 1;
+    oneof payload {
+        InitialPlacement initial_placement_payload = 2;
```

Comment:
I would remove the trailing `payload` from all of these. The variable is accessed via `actionSpecification.payload.initial_placement` so no reason to add another `payload` at the end of the name, right?

Reply:
so in scala at least what this results to is a class with type `InitialPlacementPayload` so if the variable is just named `initial_placement` it would result in a class called `InitialPlacment` which conflicts with the other `InitialPlacement` class. Its not too much of a problem and can be worked around if you think it would be better. But it was much more cinviennet to have it be `initial_placement_payload`

Reply:
Gotcha, sounds good to me 👍